src/traces/surface/convert.js

```diff
@@ -434,7 +434,7 @@ proto.update = function(data) {
     var sceneLayout = scene.fullSceneLayout;
     var surface = this.surface;
     var alpha = data.opacity;
-    var colormap = parseColorScale(data, alpha);
+    var colormap = parseColorScale(data);
```

Comment:
Does `trace.opacity` still do anything with these modifications? I'd expect it to effectively multiply the opacity of the entire colorscale.

And do we need any adjustments down here:
https://github.com/plotly/plotly.js/blob/13324e220e22d1f80cb386c4df5af894b18e3996/src/traces/surface/convert.js#L683-L686
?

Reply:
> Does `trace.opacity` still do anything with these modifications? I'd expect it to effectively multiply the opacity of the entire colorscale.

Positive. We have mocks for that. Also tested interactively.

Reply:
> And do we need any adjustments down here:
> https://github.com/plotly/plotly.js/blob/13324e220e22d1f80cb386c4df5af894b18e3996/src/traces/surface/convert.js#L683-L686
> 
> ?

Good call. Adjustments in https://github.com/plotly/plotly.js/pull/5166/commits/cfe9dd1fd7f1712eaab217210b5a665e4a7fbd50.

So this PR should be ready to go.